src/Param.cc

```diff
@@ -613,10 +719,43 @@ bool Param::SetFromString(const std::string &_value)
   return this->dataPtr->set;
 }
 
+//////////////////////////////////////////////////
+bool Param::SetFromString(const std::string &_value)
+{
+  return this->SetFromString(_value, false);
+}
+
+//////////////////////////////////////////////////
+ElementPtr Param::GetParentElement() const
+{
+  return this->dataPtr->parentElement.lock();
+}
+
+//////////////////////////////////////////////////
+void Param::SetParentElement(const ElementPtr _parentElement)
+{
+  if (!_parentElement)
+    this->dataPtr->parentElement.reset();
+
+  this->dataPtr->parentElement = _parentElement;
+
+  if (this->dataPtr->strValue.has_value())
+  {
+    const std::string strVal = this->dataPtr->strValue.value();
+    if (!this->SetFromString(strVal))
+    {
+      sdferr << "Failed to set value [" << strVal
+          << "] for new parent element of " << "name ["
+          << _parentElement->GetName() << "], reverting to previous value.\n";
```

Comment:
Can you check if `_parentElement` is `nullptr`?

Reply:
Thanks for catching that, it would have been disastrous, https://github.com/ignitionrobotics/sdformat/pull/589/commits/e44c8409affed3c9e75d55b70e1076c29aa7d9f0